### ISSUE TITLE
Vertical Tabs Crash

### ISSUE THREAD

**Issue by Comprehensive-Jason** [brave/brave-browser#29398]
## Description 
Brave Nightly crashes as soon as vertical tabs are enabled. 

## Steps to Reproduce

   1. Open Brave Nightly
   2. Right click on tab and select **use vertical tabs**

## Actual result:
Brave Nightly immediately crashes

## Expected result:
Brave should not crash and vertical tabs should work as they have been in previous releases.

## Reproduces how often: 
Easily reproduced, occurs every time

## Brave version (brave://version info)
|                       |                                                                                                                                      |
|  ------------   |                 --------------------------------------------------------------------                  |
|  Brave    	|  1.52.2 Chromium: 112.0.5615.39 (Official Build) nightly (64-bit)                                |
|  Revision	|  a0e7b9718a92bcd1cf33b7c95316caff3fc20714-refs/branch-heads/5615@{#753}    |
|  OS	        |  Windows 11 Version 22H2 (Build 22624.1470)                                                           |

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
No, vertical tabs aren't rolled out yet.
- Can you reproduce this issue with the beta channel?
N/A, don't use. 
- Can you reproduce this issue with the nightly channel? 
Yes, every time.

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
No.
- Does the issue resolve itself when disabling Brave Rewards? 
No.
- Is the issue reproducible on the latest version of Chrome? 
No, vertical tabs aren't available on Chrome.

## Miscellaneous Information:
This issue just started occurring 1 day ago.

**Comment by rebron** [comment]
@Comprehensive-Jason Do you see a report in brave://crashes? Can you send and post the crash id here.

**Comment by rebron** [comment]
cc: @sangwoo108

**Comment by Comprehensive-Jason** [comment]
@rebron This is what I got:

Crash from Wednesday, March 29, 2023 at 7:48:17 PM
Status:	Upload requested by user
Local Crash Context:	e5f62648-4288-4a54-86c5-fa082efbfad3

**Comment by Comprehensive-Jason** [comment]
Uploaded:

Crash from Wednesday, March 29, 2023 at 7:48:17 PM
Status:	Uploaded
Uploaded Crash Report ID:	2daa0000-0385-da09-0000-000000000000
Upload Time:	Wednesday, March 29, 2023 at 8:03:28 PM
Local Crash Context:	e5f62648-4288-4a54-86c5-fa082efbfad3

**Comment by sangwoo108** [comment]
Thank you so much for reporting.

Looks like this one

```
[ 00 ] `anonymous namespace'::BraveVerticalTabStyle::GetPath(TabStyle::PathType,float,bool,PathType::RenderUnits) ( brave_tab_style_views.inc.cc:154 )
[ 01 ] _tailMerge_esent.dll
[ 02 ] _tailMerge_esent.dll
[ 03 ] static void `anonymous namespace'::GM2TabStyle::PaintTabBackground(class gfx::Canvas *, TabActive, class absl::optional<int>, int) ( tab_style_views.cc:823 )
[ 04 ] _tailMerge_esent.dll
[ 05 ] ThemeServiceFactory::GetBrowserContextToUse(content::BrowserContext *) ( theme_service_factory.cc:27 )
[ 06 ] _tailMerge_esent.dll
[ 07 ] malloc ( allocator_shim_override_ucrt_symbols_win.h:71 )
[ 08 ] Profile::IsTor() ( profile.cc:24 )
[ 09 ] profile_metrics::GetBrowserProfileType(base::SupportsUserData const *) ( browser_profile_type.cc:47 )
[ 10 ] `anonymous namespace'::GM2TabStyle::PaintTab(gfx::Canvas *) ( tab_style_views.cc:515 )
[ 11 ] SkCanvas::SkCanvas(SkIRect const &) ( SkCanvas.cpp:474 )
[ 12 ] ui::PaintContext::PaintContext(ui::PaintContext const &,gfx::Vector2d const &) ( paint_context.cc:33 )
[ 13 ] SkCanvas::internalRestore() ( SkCanvas.cpp:1353 )
[ 14 ] `anonymous namespace'::BraveVerticalTabStyle::PaintTab(gfx::Canvas *) ( brave_tab_style_views.inc.cc:171 )
[ 15 ] views::View::Paint(views::PaintInfo const &) ( view.cc:1243 )
[ 16 ] void GlyphTrackingDevice::~GlyphTrackingDevice() ( SkChromeRemoteGlyphCache.cpp:642 )
[ 17 ] _tailMerge_esent.dll
[ 18 ] _tailMerge_esent.dll
[ 19 ] _tailMerge_esent.dll
[ 20 ] _tailMerge_esent.dll
```

**Comment by sangwoo108** [comment]
@Comprehensive-Jason , Sorry but I wasn't able to reproduce this bug on my Win10/11. Could you tell me more about your tabs? Do you have pinned tabs or groups? If so, how many are they? 

And also I'm wondering if this happens when you have a single new tab.

**Comment by Comprehensive-Jason** [comment]
I am using a fresh install of Brave Nightly. I previously tried uninstalling it and clearing my browsing data to fix the problem, but the crashing did not stop. Here's a video of the crash behavior:
![NIghtly Crash](https://user-images.githubusercontent.com/68677082/228855618-228c2323-b32d-405f-ab4f-ffabe4347ffd.gif)

**Comment by sangwoo108** [comment]
@simonhong @petemill @fallaciousreasoning , could you help me with this? Honestly, I have no idea what can cause this in the call stack.

**Comment by fallaciousreasoning** [comment]
Interesting - I can't reproduce either (not in a new profile, not in my existing Nightly). The stacktrace doesn't look particularly helpful either - do you think it could be hitting the `CHECK_EQ` [here](https://github.dev/brave/brave-core/blob/3c22732081077d2fb574b00d72aa30643038fd5a/browser/ui/views/tabs/brave_tab_style_views.inc.cc#L108) and just not giving us some useful output?

I did encounter a similar (ish) issue a while back on Linux, where the browser would crash when creating a new window with vertical tabs enabled but it seems different enough that it could be unrelated.

@Comprehensive-Jason do you mind trying the even more drastic step of (temporarily) deleting your Nightly profile?
1. Open brave://version/ and find your `Profile Path`
![image](https://user-images.githubusercontent.com/7678024/229006889-f990bc29-ada5-49ef-a035-4cb2f5cbe3e5.png)
2. Close Nightly
3. Rename the `Brave-Browser-Nightly` folder to `Brave-Browser-Nightly.bak` (mine is in `C:\Users\jayha\AppData\Local\BraveSoftware\`)
4. Open Nightly and test if it still crashes when you enable vertical tabs

To get back your old profile:
1. Close Nightly
2. Delete the new `Brave-Browser-Nightly` folder
3. Rename `Brave-Browser-Nightly.bak` to `Brave-Browser-Nightly` (or if you want to be extra sure you don't lose it, copy it and then rename).

**Comment by sangwoo108** [comment]
> do you think it could be hitting the CHECK_EQ [here](https://github.dev/brave/brave-core/blob/3c22732081077d2fb574b00d72aa30643038fd5a/browser/ui/views/tabs/brave_tab_style_views.inc.cc#L108) and just not giving us some useful output?

Yeah, sounds possible. I'll try adjusting DPI on my machine.

**Comment by sangwoo108** [comment]
Oh, I adjusted dpi to 175% and it crashed. @Comprehensive-Jason  Could you checkout your DPI? I guess it might be okay with 150% or 200%. Meanwhile, I'll try fixing it.

**Comment by Comprehensive-Jason** [comment]
@fallaciousreasoning I tried your suggestion. The crash still occurs.
@sangwoo108 I changed Windows scale to 300% from my default 125%. **The crash no longer occurs!**

**Comment by sangwoo108** [comment]
300% thanks! Will chekc that too

**Comment by Comprehensive-Jason** [comment]
@sangwoo108 After changing back to 125%, it now crashes immediately on startup, and I can't open it at all.

**Comment by sangwoo108** [comment]
100% might be save you to open it

**Comment by Comprehensive-Jason** [comment]
100, 150 percent work. 125,175 percent don't work.

**Comment by sangwoo108** [comment]
Thanks for checking! Will make a patch asap. Thanks @fallaciousreasoning , That was a huge help!

**Comment by kjozwiak** [comment]
The above requires `1.51.86` or higher for `1.51.x` verification 👍

**Comment by stephendonner** [comment]
## Verification `PASSED` using 


Brave | 1.51.86 Chromium: 112.0.5615.49 (Official Build) beta (64-bit)
-- | --
Revision | bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}
OS | Windows 10 Version 22H2 (Build 19045.2846)


### First, reproduced original crash using `1.52.2`:
<details>

### Steps:
1. on `Windows 10`, opened the `Display` control-panel applet
2. chose `175% / 125%` (see screenshot, above) for `Scale and layout` / `Custom scaling` values
3. launched Brave
4. loaded a few sites
5. context-clicked on the tab's title
6. chose `Use vertical tabs`

Crashed, in (sample incident ID of [2d380200-cefb-e309-0000-000000000000](https://share.backtrace.io/api/share/mWkVkZ1ZL7gQdKBfyzK8q2)):

```
[ 00 ] `anonymous namespace'::BraveVerticalTabStyle::GetPath(TabStyle::PathType,float,bool,PathType::RenderUnits) ( brave_tab_style_views.inc.cc:154 )
[ 01 ] _tailMerge_esent.dll
[ 02 ] _tailMerge_esent.dll
[ 03 ] static void `anonymous namespace'::GM2TabStyle::PaintTabBackground(class gfx::Canvas *, TabActive, class absl::optional<int>, int) ( tab_style_views.cc:823 )
[ 04 ] static void * allocator_shim::internal::PartitionMalloc(const struct allocator_shim::AllocatorDispatch *, unsigned __int64, void *) ( allocator_shim_default_dispatch_to_partition_alloc.cc:302 )
[ 05 ] _tailMerge_esent.dll
[ 06 ] ThemeServiceFactory::GetBrowserContextToUse(content::BrowserContext *) ( theme_service_factory.cc:27 )
[ 07 ] _tailMerge_esent.dll
[ 08 ] _tailMerge_esent.dll
[ 09 ] malloc ( allocator_shim_override_ucrt_symbols_win.h:71 )
[ 10 ] Profile::IsTor() ( profile.cc:24 )
[ 11 ] profile_metrics::GetBrowserProfileType(base::SupportsUserData const *) ( browser_profile_type.cc:47 )
[ 12 ] `anonymous namespace'::GM2TabStyle::PaintTab(gfx::Canvas *) ( tab_style_views.cc:515 )
[ 13 ] _tailMerge_esent.dll
[ 14 ] `anonymous namespace'::BraveVerticalTabStyle::PaintTab(gfx::Canvas *) ( brave_tab_style_views.inc.cc:171 )
[ 15 ] views::View::Paint(views::PaintInfo const &) ( view.cc:1243 )
[ 16 ] void GlyphTrackingDevice::~GlyphTrackingDevice() ( SkChromeRemoteGlyphCache.cpp:642 )
[ 17 ] _tailMerge_esent.dll
[ 18 ] _tailMerge_esent.dll
[ 19 ] _tailMerge_esent.dll
[ 20 ] _tailMerge_esent.dll
```


example | example | example
----------|-----------|----------
![image (1)](https://user-images.githubusercontent.com/387249/230559849-3e81235d-10f6-44fc-86bc-cf2e112c26ac.png) | ![image](https://user-images.githubusercontent.com/387249/230559852-5713a12a-b358-4d4b-8d01-4c4efedd056d.png) |  ![image](https://user-images.githubusercontent.com/387249/230559876-99dac272-3de0-43c0-b9f7-88ca0474dc6a.png)

</details>

### Confirmed FIXED using `1.51.86`:

### Steps:
1. on `Windows 10`, opened the `Display` control-panel applet
2. chose `175% / 125%` (see screenshot, above) for `Scale and layout` / `Custom scaling` values
3. launched Brave
4. loaded a few sites
5. context-clicked on the tab's title
6. chose `Use vertical tabs`
7. confirmed no crash
8. shut down and relaunched
9. confirmed no crash
10. drag & dropped tabs
11. confirmed no crash

example | example
----------|----------

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using
```
Brave | 1.51.86 Chromium: 112.0.5615.49 (Official Build) beta (64-bit)
-- | --
Revision | bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}
OS | Windows 11 Version 22H2 (Build 22621.1555)
```


### Steps:
1. on `Windows 11`, opened the `Display` control-panel applet
2. chose `175% / 125%` (see screenshot, above) for `Scale and layout` / `Custom scaling` values
3. launched Brave
4. loaded a few sites
5. context-clicked on the tab's title
6. chose `Use vertical tabs`
7. confirmed no crash
8. shut down and relaunched
9. confirmed no crash
10. drag & dropped tabs
11. confirmed no crash

#### 150% (Recommended)

ex|ex|ex
---|---|----
<img width="670" alt="image" src="https://user-images.githubusercontent.com/98358127/231536782-32a55636-9593-4292-a23a-bb581d4ceb09.png">|<img width="670" alt="image" src="https://user-images.githubusercontent.com/98358127/231536830-ce592f66-a04c-41dc-b8c7-60252f4c41e7.png">|<img width="670" alt="image" src="https://user-images.githubusercontent.com/98358127/231536989-eae4354c-34c8-4155-bf97-6c24384982ae.png">


#### 125%
ex|ex|ex
---|---|----
<img width="670" alt="image" src="https://user-images.githubusercontent.com/98358127/231585673-f142727b-2b23-42f1-bd51-6ac78b834eb9.png">|<img width="670" alt="image" src="https://user-images.githubusercontent.com/98358127/231585544-2c29a525-b549-46b6-a952-9010340070a8.png">|<img width="970" alt="image" src="https://user-images.githubusercontent.com/98358127/231586058-1d46ecb6-329b-414e-a239-b1e656b48774.png">|

#### 175 %

ex|ex|ex
---|---|----
<img width="670" alt="image" src="https://user-images.githubusercontent.com/98358127/231586635-25dd23ad-9ea9-4074-8077-b6df60d1ed55.png">|<img width="670" alt="image" src="https://user-images.githubusercontent.com/98358127/231576473-5d7a8a1f-5790-4544-9ec4-a103d7d9d828.png">|<img width="970" alt="image" src="https://user-images.githubusercontent.com/98358127/231587192-b736825c-f124-42b7-af0a-a3dc06422074.png">

#### 200%

ex|ex|ex
---|---|----
<img width="670" alt="image" src="https://user-images.githubusercontent.com/98358127/231577381-9603ea05-d5a1-4851-8f81-68e9da842d0c.png">|<img width="670" alt="image" src="https://user-images.githubusercontent.com/98358127/231580909-41043de9-b272-4777-aeb2-e9e27495b5c8.png">|<img width="970" alt="image" src="https://user-images.githubusercontent.com/98358127/231580887-0e7da2e2-269c-4e88-ab80-df7c7cf38381.png">
